docs/adr/0003-workspace-blob-caching.md

```diff
@@ -0,0 +1,56 @@
+# 3. Workspace BLOB Caching
+
+* Status: [ **proposed** | rejected | accepted | deprecated ]
+* Date: 2023-09-20 
+* Authors: @chanwit
+* Deciders: TBD 
+
+## Context
+
+The TF-Controller currently faces challenges related to the deletion of Terraform resources.
+These problems span across three categories:
+
+1. Single object deletion,
+2. Resources with dependencies deletion, and
+3. Namespace deletion.
+
+These problems must be fixed in the above order as (2) and (3) require single object deletion to be resolved first.
+
+Deleting a single TF object can sometimes be obstructed because it's tied to other resources like Source objects, Secrets, and ConfigMaps. If we try to remove it without deleting these resources, the TF object gets stuck in an inconsistent state, making it harder for users to manage their infrastructure smoothly.
+Therefore, the TF-Controller is being enhanced to address this problem more efficiently, using the contents of generated Workspace BLOBs. Each BLOB contains all necessary information from the associated Source, Secrets, and ConfigMaps to ensure that TF-Controller finalization procedures can delete objects correctly.
```

Comment:
What is it more efficient than?

If I'm a user that has got into this situation of having an undeletable Terraform object, is there anything I can do to unstick it -- presumably, recreating the missing resources would work, right? If so, I think you should mention that here or as part of the explanation of the problem; and, say that the proposed solution makes it much less likely that a user will have to step in and fix things themselves. (That really would be more efficient!)